### ISSUE TITLE
Should the `missing_const_for_fn` lint be less applicable for private functions never used in consts?

### ISSUE THREAD

**Issue by LiamTheProgrammer** [rust-lang/rust-clippy#6901]
So if there is a function that provably is never used in a constant based on its visibility, then why should it be suggested to mark it as `const`? Shouldn't the lint doing that only suggest changing the function to a `const fn` when it's useful to do it?

Example (yes the result just be declared as a `const` but that's a different subject):

```rust
fn marked_as_should_be_const() -> u64 { 
    0 
}
```

If this was a public function, then yes it should be constant to make it more flexible, but if it's only used internally then there's no need to until it's required. 

Possible fix PR:

`rustc` itself would give an error complaining that it isn't a `const fn` already if it was internally used, and so this lint can simply be skipped entirely for private functions, which should be pretty simple to implement.

**Comment by llogiq** [comment]
The downside is that being const requires all called functions being const. Say `pub fn a()` calls `fn b()` that could be const but isn't, and `a` could be const if not for the call of `b`, we could not give a machine applicable suggestion because we have no way of declaring suggestion dependencies.

**Comment by LiamTheProgrammer** [comment]
Ah. Good point.
